pkg/messaging/mqtt/pubsub.go

```diff
@@ -21,13 +21,14 @@ const (
 )
 
 var (
-	errConnect                = errors.New("failed to connect to MQTT broker")
-	errSubscribeTimeout       = errors.New("failed to subscribe due to timeout reached")
-	errUnsubscribeTimeout     = errors.New("failed to unsubscribe due to timeout reached")
-	errUnsubscribeDeleteTopic = errors.New("failed to unsubscribe due to deletion of topic")
-	errNotSubscribed          = errors.New("not subscribed")
-	errEmptyTopic             = errors.New("empty topic")
-	errEmptyID                = errors.New("empty ID")
+	ErrConnect                = errors.New("failed to connect to MQTT broker")
+	ErrSubscribeTimeout       = errors.New("failed to subscribe due to timeout reached")
+	ErrUnsubscribeTimeout     = errors.New("failed to unsubscribe due to timeout reached")
+	ErrUnsubscribeDeleteTopic = errors.New("failed to unsubscribe due to deletion of topic")
+	ErrNotSubscribed          = errors.New("not subscribed")
+	ErrEmptyTopic             = errors.New("empty topic")
+	ErrEmptyID                = errors.New("empty ID")
+	ErrFailed                 = errors.New("failed")
```

Comment:
What has failed?
I have seen an instance where you are using if after failing to handle mainflux message. Probably change it to something specific i.e
```go
	ErrFailedToHandleMessage  = errors.New("failed to handle mainflux message")

```